provider/rfc2136/rfc2136.go

```diff
@@ -209,30 +216,32 @@ func (r rfc2136Provider) List() ([]dns.RR, error) {
 		return make([]dns.RR, 0), nil
 	}
 
-	log.Debugf("Fetching records for '%s'", r.zoneName)
+	records := make([]dns.RR, 0)
+	for _, zone := range r.zoneNames {
+		log.Debugf("Fetching records for '%s'", zone)
 
-	m := new(dns.Msg)
-	m.SetAxfr(r.zoneName)
-	if !r.insecure && !r.gssTsig {
-		m.SetTsig(r.tsigKeyName, r.tsigSecretAlg, clockSkew, time.Now().Unix())
-	}
+		m := new(dns.Msg)
+		m.SetAxfr(dns.Fqdn(zone))
+		if !r.insecure && !r.gssTsig {
+			m.SetTsig(r.tsigKeyName, r.tsigSecretAlg, clockSkew, time.Now().Unix())
+		}
 
-	env, err := r.actions.IncomeTransfer(m, r.nameserver)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch records via AXFR: %v", err)
-	}
+		env, err := r.actions.IncomeTransfer(m, r.nameserver)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch records via AXFR: %v", err)
```

Comment:
%v -> %w
That's for all return errors

Reply:
fixed: https://github.com/kubernetes-sigs/external-dns/pull/3976/commits/0ac4740162c64059a8557f5226b2609f43c20f99